web/src/components/Settings/WorkspaceSection.tsx

```diff
@@ -128,7 +128,10 @@ const WorkspaceSection = () => {
       <div className="w-full flex flex-row justify-between items-center">
         <span>{t("setting.workspace-section.disallow-password-auth")}</span>
         <Switch
-          disabled={workspaceStore.state.profile.mode === "demo" || identityProviderList.length === 0}
+          disabled={
+            workspaceStore.state.profile.mode === "demo" ||
+            (identityProviderList.length === 0 && !workspaceGeneralSetting.disallowPasswordAuth)
```

Comment:
Disabling password login requires setting up SSO first; otherwise, there would be no way to log in. So I think the logic here is correct.

Reply:
Don't you agree that one should be able to enable password login regardless?

I reproduced the erroneous scenario and mentioned it in the issue:
1- Add an OAuth login
2- Disallow passwords
3- Remove OAuth
4- You can no longer enable password login as the switch is always disabled

Reply:
Gocha